Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM rust:1.68-slim-buster as builder
+WORKDIR /app/src
+
+# Force crates.io init for better docker caching
+COPY docker/caching.rs src/main.rs
+COPY Cargo.lock .
+COPY Cargo.toml .
+RUN cargo build --release
+
+COPY . .
+RUN cargo build --release
+
+EXPOSE 3100
+ENTRYPOINT [ "/app/src/target/release/agartex-authentication" ]
```

Comment:
Almost forgot
```suggestion
FROM debian:10.13-slim

WORKDIR /app
RUN chmod 777 .

RUN useradd user
USER user

COPY --from=builder /app/src/target/release/agartex-authentication .

EXPOSE 3100
ENTRYPOINT [ "./agartex-authentication" ]
```

Reply:
What's the difference?

Reply:
Generaly, you don't need to keep all the files used in the building stage, but only the executable. It allows to reduce the size of the docker image. See https://docs.docker.com/build/building/multi-stage/